src/codegen/generators/expression_generator.rs

```diff
@@ -855,17 +888,62 @@ impl<'ink, 'b> ExpressionCodeGenerator<'ink, 'b> {
                 self.generate_call_struct_argument_assignment(&CallParameterAssignment {
                     assignment_statement: exp,
                     function_name,
-                    parameter_type: None,
                     index: index as u32,
                     parameter_struct,
                 })?;
             if let Some(parameter) = parameter {
                 result.push(parameter.into());
             };
         }
+
         Ok(result)
     }
 
+    fn get_parameter_type(&self, parameter: &VariableIndexEntry) -> String {
+        if let Some(DataTypeInformation::Pointer {
+            inner_type_name,
+            auto_deref: true,
+            ..
+        }) = self
+            .index
+            .find_effective_type_info(parameter.get_type_name())
+        {
+            inner_type_name.into()
+        } else {
+            parameter.get_type_name().into()
+        }
+    }
+
+    fn generate_empty_expression(
+        &self,
+        ty: BasicTypeEnum<'ink>,
+        parameter: &VariableIndexEntry,
+    ) -> Result<BasicValueEnum<'ink>, Diagnostic> {
+        let name = "empty_varinout";
+        match parameter.get_declaration_type() {
+            ArgumentType::ByVal(..) => {
+                if let Some(initial_value) = self
+                    .index
+                    .get_const_expressions()
+                    .maybe_get_constant_statement(&parameter.initial_value)
+                {
+                    self.generate_expression(initial_value)
+                } else {
+                    let load_name = format!(
+                        "{}{}{}",
+                        self.temp_variable_prefix, name, self.temp_variable_suffix
+                    );
+                    let ptr_value = self.llvm.builder.build_alloca(ty, name);
+                    Ok(self.llvm.load_pointer(&ptr_value, load_name.as_str()))
```

Comment:
should we change the name for build_alloca also to empty ?
i thought it would probably make sense when checking the IR to see where it comes from but it's ok for me if we change it to empty